content/docs/intro/languages/dotnet.md

```diff
@@ -0,0 +1,94 @@
+
+```bash
+$ mkdir myproject && cd myproject
+$ pulumi new csharp
+```
+
+The template will autogenerate a set of files that make up a Pulumi project.
+
+{{% lang csharp %}}
+
+## C\#
+
+You can write Pulumi programs in C# to get additional verification and tooling benefits. As of version 1.5, Pulumi supports .NET natively. The fastest way to get started with Pulumi in C#, is to use a template:
+
+```bash
+$ mkdir myproject && cd myproject
+$ pulumi new csharp
+```
+
+This will create a `Pulumi.yaml` [project file]({{< relref "project.md" >}}) containing some minimal metadata about your project (including a name and description which you may wish to change), a `Infra.csproj` file that holds references used by the project, and a `Program.cs` file, containing your program. The `.csproj` file can be named more appropriately depending upon the project. The name of the directory is used as the project name in `Pulumi.yaml`.
+
+In addition to the C# template, there are additional .NET templates for cloud providers. To list the templates:
+
+```bash
+$ pulumi new
+```
+{{% /lang %}}
+
+
+{{% lang fsharp %}}
+
+## F\#
+
+You can write Pulumi programs in F# to get additional verification and tooling benefits. The fastest way to get started with Pulumi in F# is to use a template:
+
+```bash
+$ mkdir myproject && cd myproject
+$ pulumi new fsharp
+```
+
+This will create a `Pulumi.yaml` [project file]({{< relref "project.md" >}}) containing some minimal metadata about your project (including a name and description which you may wish to change), a `Infra.fsproj` file that holds references used by the project, and a `Program.fs` file, containing your program. The `.fsproj` file can be named more appropriately depending upon the project. The name of the directory is used as the project name in `Pulumi.yaml`.
+
+In addition to the F# template, there are additional .NET templates for cloud providers. To list the templates:
+
+```bash
+$ pulumi new
+```
+
+{{% /lang %}}
+
+{{% lang visualbasic %}}
+
+## Visual Basic
+
+You can write Pulumi programs in Visual Basic to get additional verification and tooling benefits. The easiest way to get started with Pulumi in Visual Basic is to use a template:
+
+```bash
+$ mkdir myproject && cd myproject
+$ pulumi new visualbasic
+```
+
+This will create a `Pulumi.yaml` [project file]({{< relref "project.md" >}}) containing some minimal metadata about your project (including a name and description which you may wish to change), a `Infra.vbproj` file that holds references used by the project, and a `Program.vb` file, containing your program. The `.vbproj` file can be named more appropriately depending upon the project. The name of the directory is used as the project name in `Pulumi.yaml`.
+
+In addition to the Visual Basic template, there are additional .NET templates for cloud providers. To list the templates:
+
+```bash
+$ pulumi new
+```
+
+{{% /lang %}}
```

Comment:
Key things I think we should mention that are not yet called out here:
1. Pulumi packages are in Nuget - link to them in nuget.org.
2. Pulumi automatically runs `dotnet build` as part of deployments.
3. Highlight that you can use Visual Studio Code and/or Visual Studio to get full tooling support for Pulumi with .NET (including screenshots).
4. Add a note that .NET support is in preview, and under active development.

Reply:
accepted